temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInternal.java

```diff
@@ -188,11 +189,23 @@ public static long currentTimeMillis() {
       Class<T> activityInterface,
       ActivityOptions options,
       Map<String, ActivityOptions> activityMethodOptions) {
+    // Merge the activity options we may have received from the workflow with the options we may
+    // have received in WorkflowImplementationOptions.
+    Map<String, ActivityOptions> mergedActivityOptionsMap = new Hashtable<>();
```

Comment:
Use HashMap instead.

https://www.geeksforgeeks.org/differences-between-hashmap-and-hashtable-in-java/